tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/annotations/annotation.bal

```diff
@@ -116,6 +116,52 @@ function testAnnotationsOnTypeRefTypes() {
     assertTrue('annotation is ());
 }
 
+@foo:annot {
+    examples: foo:EXAMPLES
+}
+type Student record {|
+    int id;
+    string name;
+|};
+
+public const foo:Examples EXAMPLES_2 = {
+    response: {}
+};
+
+@foo:annot {
+    examples: EXAMPLES_2
+}
+type Employee record {|
+    int id;
+    string name;
+|};
+
+function testConstAnnotationAccess() {
+    Employee employee = {id: 1, name: "chirans"};
+    typedesc<any> t = typeof employee;
+    foo:AnnotationRecord? annot = t.@foo:annot;
+    assertTrue(annot is foo:AnnotationRecord);
+    foo:AnnotationRecord config = <foo:AnnotationRecord> annot;
+    assertEquality({"response":{}}, config.examples);
+    assertTrue(config.examples is readonly);
+
+    Student student = {id: 2, name: "sachintha"};
+    typedesc<any> s = typeof student;
+    foo:AnnotationRecord? annot1 = s.@foo:annot;
+    assertTrue(annot1 is foo:AnnotationRecord);
+    foo:AnnotationRecord config1 = <foo:AnnotationRecord> annot1;
+    assertEquality({"response":{}}, config1.examples);
+    assertTrue(config1.examples is readonly);
+
+    foo:Teacher teacher = {id: 1, name: "James"};
+    typedesc<any> t1 = typeof teacher;
+    foo:AnnotationRecord? annot2 = t1.@foo:annot;
+    assertTrue(annot2 is foo:AnnotationRecord);
+    foo:AnnotationRecord config2 = <foo:AnnotationRecord> annot2;
+    assertEquality({"response":{}}, config2.examples);
```

Comment:
At least for one of these shall we add a field or two in response?

Reply:
Updated in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/43163/commits/7d7035e087f093e437262a77f6b46597519d622e).